capsules/src/nonvolatile_to_pages.rs

```diff
@@ -46,17 +46,17 @@ enum State {
     Write,
 }
 
-pub struct NonvolatileToPages<'a, F: hil::flash::Flash + 'static> {
+pub struct NonvolatileToPages<'driver, 'client, 'cbuf, F: hil::flash::Flash + 'static> {
```

Comment:
Ditto here: the lifetimes should be able to all be the same, no? The driver and client _definitely_ need always be equal (you'll never be able to instantiate something where the driver and client are not the same), and the buffer should always live _at least_ that long.

Reply:
I don't (wasn't required up to now to) have much experience with lifetimes, and yes, of course we can use upcasting here.
So having one lifetime `'a` should allow to use it like `'driver`, `'client: 'driver` and `'cbuf: 'client`, am I right?